code/ship/ship.cpp

```diff
@@ -17382,6 +17383,15 @@ SCP_string ship_return_orders(ship* sp)
 		}
 		break;
 
+	case AI_GOAL_CHASE_SHIP_TYPE:
```

Comment:
You can actually stack the goals here since they use exactly the same code.  Put them on consecutive lines:
```C++
case AI_GOAL_CHASE_SHIP_CLASS:
case AI_GOAL_CHASE_SHIP_TYPE:
```
Then you only need one copy of the "any"/"no orders" code